src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -196,13 +194,7 @@ public static boolean isCombinedFileIngest(FlightContext flightContext) {
         Objects.requireNonNullElse(
             flightContext.getWorkingMap().get(IngestMapKeys.NUM_BULK_LOAD_FILE_MODELS, Long.class),
             0L);
-    if (numFiles == 0) {
-      Logger logger = LoggerFactory.getLogger(flightContext.getFlightClassName());
-      logger.info(
-          "Skipping {} because there are no files to ingest", flightContext.getStepClassName());
```

Comment:
Why get rid of a more meaningful error message?

Reply:
This is double logging the error message, and it's generating the error during flight construction as well as during step execution. Since flights can be constructed at any point (e.g. when resuming a stopped flight, the entire flight will be reconstructed) it's not useful to log during flight construction.

With this change, the logs will show both the step that's skipped and the containing (optional) step that skipped it. Given both those things you can figure out why a step was skipped.

On the other hand, it would be easy to add a `getReason()` API to `OptionalStep` so that a subclass could override to return a more specific reason, if we think it would be helpful.

Reply:
Also in its original use both the positive and negative states were used; this log was assuming that only the positive state was used.

Reply:
I've implemented `OptionalStep.getReason()` as proposed and moved this message there.